backend/lib/endpoints/posts.js

```diff
@@ -418,41 +504,58 @@ async function routes(app) {
     "/:postId/comments/:commentId",
     { preValidation: [app.authenticate], schema: deleteCommentSchema },
     async (req) => {
-      const { params, userId } = req;
-      const { commentId, postId } = params;
-      const { ok, deletedCount } = await Comment.deleteMany({
-        $or: [
-          { _id: commentId, authorId: userId, postId },
-          { parentId: commentId, postId },
-        ],
-      });
-      if (ok !== 1 || deletedCount < 1) {
-        return new httpErrors.BadRequest();
+      const { userId } = req.body;
+      const { commentId } = req.params;
+      const [findErr, comment] = await app.to(Comment.findById(commentId));
+      if (findErr) {
+        throw app.httpErrors.notFound();
+      } else if (comment.author.id !== userId) {
+        throw app.httpErrors.forbidden();
+      }
+
+      const [deleteCommentErr, deletedComment] = await app.to(comment.delete());
+      if (deleteCommentErr) {
+        req.log.error("Failed deleting comment", { deleteCommentErr });
+        throw app.httpErrors.internalServerError();
       }
-      return { deletedCount, success: true };
+
+      const {
+        deletedCount: deletedNestedCount,
+        ok: deleteNestedOk,
+      } = await Comment.deleteMany({ parentId: commentId });
+      if (deleteNestedOk !== 1) {
+        app.log.error("failed removing nested comments for deleted comment", {
```

Comment:
You mean deleting the comment and failing for the nested?
In that case, we would have dangling comments, the same for deleting post and failing for comments.

The comments would be left dangling their parents here but wouldn't be shown/returned/counted (since we always start at base level and go up, they would never be reached).

Ideally, we would wrap this in a transaction, this way if deleting nested fails we can always abort it and revert changes, but that requires setting up the replica set.

Reply:
ok we have talked about this PR in the meeting and thought it's probably better to get this merged ASAP and integrate with frontend so it can be tested
@Naraujo13 do you think we should make a new issue for these transactions?